driver/pace/driver/initialization.py

```diff
@@ -141,13 +151,96 @@ def get_driver_state(
         self,
         quantity_factory: pace.util.QuantityFactory,
         communicator: pace.util.CubedSphereCommunicator,
+        damping_coefficients: pace.util.grid.DampingCoefficients,
+        driver_grid_data: pace.util.grid.DriverGridData,
+        grid_data: pace.util.grid.GridData,
     ) -> DriverState:
         state = _restart_driver_state(
-            self.path, communicator.rank, quantity_factory, communicator
+            self.path,
+            communicator.rank,
+            quantity_factory,
+            communicator,
+            damping_coefficients,
+            driver_grid_data,
+            grid_data,
         )
+
         return state
 
 
+@InitializerSelector.register("fortran_restart")
+@dataclasses.dataclass
+class FortranRestartConfig(Initializer):
+    """
+    Configuration for fortran restart initialization.
+    """
+
+    path: str = "."
+
+    @property
+    def start_time(self) -> datetime:
+        """Reads the last line in coupler.res to find the restart time"""
+        restart_files = os.listdir(self.path)
+        coupler_file = [fl for fl in restart_files if "coupler.res" in fl][0]
+        restart_doc = self.path + coupler_file
+        fl = open(restart_doc, "r")
+        contents = fl.readlines()
+        fl.close()
+        last_line = contents.pop(-1)
+        date = [
+            dt if len(dt) == 4 else "%02d" % int(dt) for dt in last_line.split()[:6]
+        ]
+        date_dt = datetime.strptime("".join(date), "%Y%m%d%H%M%S")
+        return date_dt
+
+    def get_driver_state(
+        self,
+        quantity_factory: pace.util.QuantityFactory,
+        communicator: pace.util.CubedSphereCommunicator,
+        damping_coefficients: pace.util.grid.DampingCoefficients,
+        driver_grid_data: pace.util.grid.DriverGridData,
+        grid_data: pace.util.grid.GridData,
+    ) -> DriverState:
+        state = _restart_driver_state(
+            self.path,
+            communicator.rank,
+            quantity_factory,
+            communicator,
+            damping_coefficients,
+            driver_grid_data,
+            grid_data,
+        )
+
+        state = _update_fortran_restart_pe_peln(state)
+
+        # TODO
+        # follow what fortran does with restart data after reading it
+        # should eliminate small differences between restart input and
+        # serialized test data
+
+        return state
+
+
+def _update_fortran_restart_pe_peln(state):
+
+    ptop = state.grid_data._vertical_data.ak[0]
+    pe = state.dycore_state.pe
+    peln = state.dycore_state.peln
+    delp = state.dycore_state.delp
+
+    np = state.dycore_state.u.np
+
+    for level in range(pe.data.shape[2]):
+        pe.data[:, :, level] = ptop + np.sum(delp.data[:, :, :level], 2)
+
+    peln.data[:] = np.log(pe.data[:])
+
+    state.dycore_state.pe = pe
+    state.dycore_state.peln = peln
+
+    return state
```

Comment:
These lines do nothing, as pe and peln are already the ones attached to dycore_state.

```suggestion
    return state
```